.github/workflows/runtests.yml

```diff
@@ -114,53 +114,39 @@ jobs:
       - run: bin/doctest --force-colors
       - run: examples/all.py -q
 
-  # -------------------- Test split 1/2 latest Python -------------- #
+  # ------------------------- Test latest Python ------------------- #
 
-  tests1-latest:
-    needs: code-quality
-
-    runs-on: ubuntu-20.04
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
-        with:
-          python-version: '3.10'
-      - run: python -m pip install --upgrade pip
-      - run: pip install mpmath
-      - run: bin/test --force-colors --split=1/2
-
-  # -------------------- Test split 2/2 latest Python -------------- #
-
-  tests2-latest:
+  tests-latest:
     needs: code-quality
 
     runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        group: [1, 2, 3, 4]
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
           python-version: '3.10'
       - run: python -m pip install --upgrade pip
-      - run: pip install mpmath
-      - run: bin/test --force-colors --split=2/2
+      - run: pip install mpmath pytest pytest-split pytest-xdist
+      - run: bin/test --split ${{ matrix.group }}/4
 
   # -------------------- Test Pyodide on node ---------------------- #
 
   test-pyodide:
     needs: code-quality
```

Comment:
For faster debugging I would remove the `needs` and then make it so that the `code-quality` job fails.